e3nn/point/message_passing.py

```diff
@@ -29,3 +31,209 @@ def message(self, x_j, k):
         if k.shape[0] == 0:  # https://github.com/pytorch/pytorch/issues/37628
             return torch.zeros(0, k.shape[1])
         return torch.einsum('eij,ej->ei', k, x_j)
+
+
+class MinimalNetwork(torch.nn.Module):
+    def __init__(self, representations, device):
+        super().__init__()
+
+        from functools import partial
+
+        from e3nn.radial import GaussianRadialModel
+        from e3nn.non_linearities.rescaled_act import swish
+        from e3nn.rsh import spherical_harmonics_xyz_cuda as rsh
+
+        assert len(representations) == 2, "Only minimal networks here"
+
+        self.device = device
+
+        self.representations = representations
+        self._find_max_l_out_in()
+        self._calculate_coupling_coefficients_and_offsets()
+
+        Rs_in = representations[0]
+        Rs_out = representations[-1]
+
+        radial_model = partial(GaussianRadialModel, max_radius=3.2, min_radius=0.7, number_of_basis=10, h=100, L=3, act=swish)
+
+        self.real_spherical_harmonics = rsh
+        self.layer = TensorPassingLayer(Rs_in, Rs_out, radial_model, self.coupling_coefficients, self.coupling_coefficients_base_offsets, self.max_l_in, self.device)
+
+    def _find_max_l_out_in(self):
```

Comment:
I'm a bit confused by this function. Why do you need to impose a max order on the user input Rs? I can understand why for the filters but `o3.selection_rule` has an argument `lmax` for setting the max filter spherical harmonics which handles this.

Reply:
It does not impose anything, it literally finds maximal values of `l_out` and `l_in` used across the network. This function is needed to determine which coupling coefficients to store, as well as to construct offsets for cuda calls.